exitstats.py

```diff
@@ -89,38 +92,72 @@ def mkdirs(name):
       if e[0] != 17:   # ignore "exists"
         raise e
 
-def postproc(dir):
-    """
-    Remove all write permissions, compute md5sums, etc
-    """
-    for f in glob.glob(dir+"*"):
-        os.chmod(f, 0444)
-    subprocess.call("find . -type f | xargs md5sum > ../manifest.tmp", shell=True, chdir=dir)
-    os.rename(dir+"/../manifest.tmp", dir+"/manifest.md5")
-    os.chmod(dir+"/manifest.md5", 0555)
-    os.chmod(dir, 0555)    # And make it immutable 
-
-logtime=(60*60)
-logf=None
-olt = -1
-olddir=""
-def getlogf(t):
-  global olt, logtime, logf, server
-  lt = int(t / logtime)*logtime
-  if lt != olt:
-    olt=lt
-    if logf: logf.close()
-    logdir=time.strftime("%Y/%m/%d/", time.gmtime(lt))
-    if olddir and olddir!=logdir:
-      postproc(olddir)
+server = ""
+one_hour = (60*60)
+LogInfo = namedtuple('LogInfo', ['name', 'f'])
+
+# Map from IP address to LogInfo
+# May include key=None if we are not using per IP logs.
+logs = {}
+log_time = -1
+def closeLogs():
+  ''' Close all log files, e.g. at the top of each hour.
+  '''
+  for k, v in logs:
+    if v.f: v.f.close()
+  logs.clear()
+
+def useLocalIP():
+  ''' Interpret local environment variable to determine whether to use
+      local IP address.
+  '''
+  env_var = os.environ.get('SIDESTREAM_USE_LOCAL_IP')
+  return env_var == 'True' or env_var == 'true' or env_var == '1'
+
+def logName(server, gm, local_ip):
+  ''' Form directory name and file name for log file.
+  '''
+  logdir= time.strftime("%Y/%m/%d/", gm)
+  ts = time.strftime("%Y%m%dT%TZ", gm)
+  if local_ip != None:
+    return logdir, "%s%s_ALL%d-%s.web100"%(server, ts ,0, local_ip)
+  else:
+    return logdir, "%s%s_ALL%d.web100"%(server, ts ,0)
+
+def openLogFile(logdir, logname):
     mkdirs(logdir)
-    logname=time.strftime("%Y/%m/%d/%%s%Y%m%dT%TZ_ALL%%d.web100", time.gmtime(lt))%(server, 0)
-    print "Opening:", logname
-    logf=open(logname, "a")
-    showkey(logf)
-  return logf
+    print "Opening:", logdir+logname
+    logf = open(logdir+logname, "a")
+    logHeader(logf)
+    # Add the entry to the logs dict.
+    return logf
+
+def getLogFile(t, local_ip=None):
+  ''' getLogFile returns the appropriate logFile for the current time
+      and local_ip address.
+      If os.environment["SIDESTREAM_USE_LOCAL_IP"] is true, then it
+      uses separate files for each local IP address.
+  '''
+  global one_hour, logs, log_time, server
+  hour_time = int(t / one_hour) * one_hour
```

Comment:
Math with time makes me sad. No need to change anything, just me being sad here.